### PR TITLE
Add vararg constructor to EnvironmentProvider

### PR DESCRIPTION
**Pull request by skrzypo987**

### CODE REVIEW

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentProvider.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import java.util.List;
 import java.util.Set;
 
+import static com.google.common.collect.ImmutableList.copyOf;
```
Comment (kokosing) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentProvider.java:24]: There are several `copyOf` methods, so I prefer not to have static import for that.

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentProvider.java [hunk 2]
```diff
@@ -32,7 +33,12 @@
 
     protected EnvironmentProvider(List<EnvironmentExtender> bases)
     {
-        this.bases = requireNonNull(bases, "bases is null");
+        this.bases = copyOf(requireNonNull(bases, "bases is null"));
+    }
+
+    protected EnvironmentProvider(EnvironmentExtender... bases)
+    {
+        this.bases = copyOf(requireNonNull(bases, "bases is null"));
```
Comment (kokosing) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentProvider.java:41]: call `this(...)` that way there is one true constructor and several convenience constructors. Then move this constructor up, so they are ordered by usage (used lower).

**Review by kokosing — APPROVED** [review]
% comments

**Comment by skrzypo987** [comment]
remarks addressed

**Review by wendigo — APPROVED** [review]

**Comment by kokosing** [comment]
Merged, thanks!

**Pull request merged**